File: CanHeEscape/sol.py
def findEscape(x:int ,y:int):
    visit[x][y] = 1

    if((x==0)or(x==n_row-1)or(y==0)or(y==n_col-1)): #check if current position is possible escape
        return f"{x},{y}"

    if(maze[x-1][y]=="1" and visit[x-1][y]==0 ): #check if element on top is free and not visited
        neighborEscape = findEscape(x-1, y)
        if(neighborEscape):
            return f"{x},{y}-"+neighborEscape

    if(maze[x][y+1]=="1" and visit[x][y+1]==0): #check if element on right is free and not visited
        neighborEscape2 = findEscape(x, y+1)
        if(neighborEscape2):
            return f"{x},{y}-"+neighborEscape2

    if(maze[x+1][y]=="1" and visit[x+1][y]==0): #check if element under is free and not visited
        neighborEscape3 = findEscape(x+1, y)
        if(neighborEscape3):
            return f"{x},{y}-"+neighborEscape3

    if(maze[x][y-1]=="1" and visit[x][y-1]==0): #check if element on left is free and not visited
        neighborEscape4 = findEscape(x, y-1)
        if(neighborEscape4):
            return f"{x},{y}-"+neighborEscape4

    return ""


#data
n_row = int(input())
n_col = int(input())

#1 for free and 0 for a wall
#input_maze = "0 1 0 1 1 1 0 1 0 1 0 1 1 0 0 1 1 0 0 1 1 0 0 1 0"
input_maze = input()
x0 = int(input())
y0 = int(input())



lst = input_maze.split()

""" maze = [[0]*n_col for _ in range(n_row)]
visit = [[0]*n_col for _ in range(n_row)]
for i in range(n_row):
    for j in range(n_col):
        maze[i][j] = lst[i*n_col+j] #0 for not visited and 1 for visited """
# second method of itterating
maze = [[lst[i+j] for j in range(n_col)] for i in range(0, n_row*n_col, n_col)]
visit = [[0 for j in range(n_col)] for i in range(0, n_row*n_col, n_col)]
print(findEscape(x0,y0))

File: CanHeEscape/sol.js
const readline = require('readline');

function findEscape(x, y, maze, visit) {
    visit[x][y] = 1;

    if (x == 0 || x == maze.length - 1 || y == 0 || y == maze[0].length - 1) {
        // check if current position is possible escape
        return `${x},${y}`;
    }

    if (maze[x - 1][y] == "1" && visit[x - 1][y] == 0) {
        // check if element on top is free and not visited
        const neighborEscape = findEscape(x - 1, y, maze, visit);
        if (neighborEscape) {
            return `${x},${y}-${neighborEscape}`;
        }
    }

    if (maze[x][y + 1] == "1" && visit[x][y + 1] == 0) {
        // check if element on right is free and not visited
        const neighborEscape2 = findEscape(x, y + 1, maze, visit);
        if (neighborEscape2) {
            return `${x},${y}-${neighborEscape2}`;
        }
    }

    if (maze[x + 1][y] == "1" && visit[x + 1][y] == 0) {
        // check if element under is free and not visited
        const neighborEscape3 = findEscape(x + 1, y, maze, visit);
        if (neighborEscape3) {
            return `${x},${y}-${neighborEscape3}`;
        }
    }

    if (maze[x][y - 1] == "1" && visit[x][y - 1] == 0) {
        // check if element on left is free and not visited
        const neighborEscape4 = findEscape(x, y - 1, maze, visit);
        if (neighborEscape4) {
            return `${x},${y}-${neighborEscape4}`;
        }
    }

    return "";
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter the number of rows in the maze: ', (n_row) => {
    rl.question('Enter the number of columns in the maze: ', (n_col) => {
        let input_maze = '';
        rl.question('Enter the maze as a string with 0s and 1s separated by spaces: ', (maze_string) => {
            input_maze = maze_string;

            rl.question('Enter the x-coordinate of the starting position: ', (x0) => {
                rl.question('Enter the y-coordinate of the starting position: ', (y0) => {

                    const lst = input_maze.split(" ");
                    const maze = Array.from({ length: n_row }, (_, i) =>
                        lst.slice(i * n_col, (i + 1) * n_col)
                    );
                    const visit = Array.from({ length: n_row }, () =>
                        Array.from({ length: n_col }, () => 0)
                    );

                    console.log("Escape path: ", findEscape(parseInt(x0), parseInt(y0), maze, visit));
                    rl.close();
                });
            });
        });
    });
});